src/horust/healthcheck.rs

```diff
@@ -181,4 +246,13 @@ file-path = "{}""#,
         assert!(healthchecks(&healthiness));
         Ok(())
     }
+
+    #[test]
+    fn test_http_healthiness_check() -> Result<()> {
+        thread::spawn(move || {
+            start_server();
+        });
+        assert_eq!(client_mock(), true);
```

Comment:
As I said elsewhere, you should not call client_mock but rather use the Healthiness function and test the `healthchecks` function.
Otherwise according to this code, you're spawning an http server and sending a request to it. It's not really testing any of code of this module.

Reply:
Hi,
I replaced the `client_mock` with `check_http_endpoint` . And moved the test related functions in the module to remove the `unused` warnings. Please take a look.

Reply:
This is definitely better, but if you have read my comment here: https://github.com/FedericoPonzi/Horust/pull/16#issuecomment-610848231 (point 2), you should call the `healthchecks` function and define a `Healthiness` struct. The `check_http_endpoint` should be called from there.

Reply:
@rajkumarGosavi please read my last comment ^